### PR TITLE
Increasing Coverage Message Processor : From 79% to 94%

### PR DESCRIPTION
**Pull request by fede-dash**
phoenix_logo_small

## Description
🧪 Test Suite Overview for message_processor

This test module provides coverage for key utility functions in message_processor.py under strands.event_loop, specifically validating behavior related to message sanitization and correction in tool use scenarios. The tests are structured with pytest and include parameterized inputs for robustness.

✅ test_clean_orphaned_empty_tool_uses

Tests the function clean_orphaned_empty_tool_uses, which removes or replaces toolUse messages that:
	•	Have empty input
	•	Do not have a matching toolResult

This ensures the message history doesn’t contain stale or incomplete tool use entries. The test checks both whether the cleanup happened and what the cleaned messages look like.

🔍 test_find_last_message_with_tool_results

Validates find_last_message_with_tool_results, which scans a message list to find the last occurrence of a toolResult. This is useful for indexing or pruning based on tool result placement.

✂️ test_truncate_tool_results

Covers truncate_tool_results, which modifies a specific message to mark its toolResult as errored (e.g., when the result is too large). The function updates the content and returns a flag indicating whether the modification occurred.

Each test ensures:
	•	Correct handling of edge cases (empty inputs, invalid indexes)
	•	No side effects on input data (using deepcopy)
	•	Expected message transformations or detection logic

## Type of Change
- Adding test coverage

Before
image 

After
image

[Choose one of the above types of changes]

## Testing
[How have you tested the change?]

* `hatch fmt --linter`
* `hatch fmt --formatter`
* `hatch test --all`
* Verify that the changes do not break functionality or introduce warnings in consuming repositories: agents-docs, agents-tools, agents-cli

## Checklist
- [X] I have read the CONTRIBUTING document
- [X ] I have added tests that prove my fix is effective or my feature works
- [X] I have updated the documentation accordingly

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/strands/event_loop/test_message_processor.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import pytest
+
+import strands.event_loop.message_processor
+from strands.event_loop import message_processor
+
+print(strands.event_loop.message_processor.__file__)
```
Comment (arronbailiss) [inline tests/strands/event_loop/test_message_processor.py]: think this print can be removed

File: tests/strands/event_loop/test_message_processor.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        # No messages
+        (
+            [],
+            False,
+            [],
+        ),
+    ],
+)
+def test_clean_orphaned_empty_tool_uses(messages, expected, expected_messages):
+    import copy
```
Comment (arronbailiss) [inline tests/strands/event_loop/test_message_processor.py]: can move this to the top of the file alongside other imports

File: tests/strands/event_loop/test_message_processor.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        (
+            [{"role": "user", "content": [{"toolResult": {"toolUseId": "1"}}]}],
+            2,
+            False,
+            [{"toolResult": {"toolUseId": "1"}}],
+        ),
+    ],
+)
+def test_truncate_tool_results(messages, msg_idx, expected_changed, expected_content):
+    import copy
```
Comment (arronbailiss) [inline tests/strands/event_loop/test_message_processor.py]: can be imported once at the top of this file

**Review by awsarron — CHANGES_REQUESTED** [review]
Thanks @fede-dash, this is great! A couple of small comments and then we'll merge this.

**Comment by fede-dash** [comment]
Ready for Review
